From 9ab30e44ded9e14e0b3a17bb7908d90309467f4c Mon Sep 17 00:00:00 2001
From: Andrew Wang <wang@apache.org>
Date: Wed, 12 Oct 2016 15:19:52 -0700
Subject: [PATCH 2478/2748] HADOOP-13700. Remove unthrown IOException from
 TrashPolicy#initialize and #getInstance
 signatures.

(cherry picked from commit 12d739a34ba868b3f7f5adf7f37a60d4aca9061b)
(cherry picked from commit f131d61ff8bc87e74c67fec3ae2b50ee50dd0d2c)
(cherry picked from commit 9bde45d2fe0f830813c6db2c7ae827937d92ab67)

Change-Id: I34e8f36cacf42630f352ab465f5dd59f79a3fdf3
---
 .../java/org/apache/hadoop/fs/TrashPolicy.java     |    6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/TrashPolicy.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/TrashPolicy.java
index 92a4d1f..1cc2aad 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/TrashPolicy.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/TrashPolicy.java
@@ -51,9 +51,8 @@
    * not assume trash always under /user/$USER due to HDFS encryption zone.
    * @param conf the configuration to be used
    * @param fs the filesystem to be used
-   * @throws IOException
    */
-  public void initialize(Configuration conf, FileSystem fs) throws IOException{
+  public void initialize(Configuration conf, FileSystem fs) {
     throw new UnsupportedOperationException();
   }
 
@@ -133,8 +132,7 @@ public static TrashPolicy getInstance(Configuration conf, FileSystem fs, Path ho
    * @param fs the file system to be used
    * @return an instance of TrashPolicy
    */
-  public static TrashPolicy getInstance(Configuration conf, FileSystem fs)
-      throws IOException {
+  public static TrashPolicy getInstance(Configuration conf, FileSystem fs) {
     Class<? extends TrashPolicy> trashClass = conf.getClass(
         "fs.trash.classname", TrashPolicyDefault.class, TrashPolicy.class);
     TrashPolicy trash = ReflectionUtils.newInstance(trashClass, conf);
-- 
1.7.9.5

